src/software/new_geom/BUILD

```diff
@@ -5,18 +5,24 @@ cc_library(
     srcs = [
         "circle.cpp",
         "point.cpp",
+        "polygon.cpp",
         "polynomial.cpp",
+        "rectangle.cpp",
         "vector.cpp",
     ],
     hdrs = [
         "angle.h",
         "angular_velocity.h",
         "circle.h",
+        "convex_polygon.h",
         "convex_shape.h",
         "geom_constants.h",
         "point.h",
+        "polygon.h",
         "polynomial.h",
         "ray.h",
+        "rectangle.h",
+        "segment.h",
         "shape.h",
         "vector.h",
```

Comment:
@jsn0716 - just a heads up on the merge conflicts this will probably cause with your PR.